### PR TITLE
chore: Update node version to 16.15.0 LTS

### LINKED ISSUE #14079: [Task]:  Update node version to latest LTS
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Update node version to the latest LTS i.e. `16.15.0` in the package.json files.
- Update the node version to the latest LTS i.e. `16.15.0` in all the CI files.

### PR DESCRIPTION
**Pull request by yaldram**
## Description
- Update the node version to v.16.15.0 in package.json and CI files.

Fixes #14079 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

- Need to test the CI and Build.
- Run test cases on the CI

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "engines": {
-    "node": "^14.15.4",
+    "node": "^16.15.0",
     "npm": "^6.14.10"
```
Comment (riodeuno) [inline app/client/package.json:7]: Could you confirm if the the relevant `npm` version should also change?

Comment (yaldram) [reply app/client/package.json]: Thanks for highlighting this I have updated the npm versions to `8.5.5`

**Comment by riodeuno** [comment]
@yaldram The build is failing with the following logs:
Screenshot 2022-05-26 at 12 51 57 PM

Have you had a chance to look into this?
I would suggest that we verify the `Run Jest Tests` Github workflow to understand why it is trying to checkout `release`.

**Comment by yaldram** [comment]
> @yaldram The build is failing with the following logs: Screenshot 2022-05-26 at 12 51 57 PM
> 
> Have you had a chance to look into this? I would suggest that we verify the `Run Jest Tests` Github workflow to understand why it is trying to checkout `release`.

Yeah I did send this pic on our slack discussion waiting for Shri to be back he can guide us better here. All the tests are passing though.

**Pull request merged**